doc/modules/neighbors.rst

```diff
@@ -514,3 +514,224 @@ the model from 0.81 to 0.82.
 
   * :ref:`sphx_glr_auto_examples_neighbors_plot_nearest_centroid.py`: an example of
     classification using nearest centroid with different shrink thresholds.
+
+
+
+.. _nca:
+
+Neighborhood Components Analysis
+================================
+
+.. sectionauthor:: William de Vazelhes <william.de-vazelhes@inria.fr>
+
+Neighborhood Components Analysis (NCA, :class:`NeighborhoodComponentsAnalysis`   )
+is a distance metric learning algorithm which aims to improve the accuracy of
+nearest neighbors classification compared to the standard Euclidean distance.
+The algorithm  directly  maximizes  a stochastic  variant  of  the
+leave-one-out k-nearest neighbors (KNN) score on the training set.  It can also
+learn a low-dimensional linear  embedding  of  labeled  data  that  can  be
+used for  data  visualization and fast classification.
+
+.. |nca_illustration_1| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_illustration_001.png
+   :target: ../auto_examples/neighbors/plot_nca_illustration.html
+   :scale: 50
+
+.. |nca_illustration_2| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_illustration_002.png
+   :target: ../auto_examples/neighbors/plot_nca_illustration.html
+   :scale: 50
+
+.. centered:: |nca_illustration_1| |nca_illustration_2|
+
+
+In the above figure, we consider some points from a randomly generated dataset.
+We focus on the stochastic KNN classification of point n°3, the thickness of a
+bond representing a softmax distance hence the weight of the neighbor vote in
+the classification. In the original space, sample 3 has many stochastic
+neighbors from various classes, so the right class is not very likely. However,
+in the embedding space, the only non-negligible stochastic neighbors are from
+the same class as sample 3, guaranteeing that the latter will be well
+classified.
+
+
+
+Classification
+--------------
+
+Combined with a nearest neighbors classifier (:class:`KNeighborsClassifier`),
+this method is attractive for classification because it can naturally
+handle multi-class problems without any increase in the model size, and no
+additional parameter than that of KNN has to be selected by the user before
```

Comment:
maybe replace by something like "and does not introduce additional parameters that require fine-tuning by the user"?

Reply:
Done.